src/towncrier/build.py

```diff
@@ -145,6 +145,18 @@ def __main(
     base_directory, config = load_config_from_options(directory, config_file)
     to_err = draft
 
+    if project_version is None:
+        project_version = config.version
+        if project_version is None:
+            if not config.package:
+                raise UsageError(
+                    "'--version' is required since the config file does "
+                    "not contain 'version' or 'package'."
+                )
+            project_version = get_version(
+                os.path.join(base_directory, config.package_dir), config.package
+            ).strip()
```

Comment:
Can't run the code rn, but we should be able to unindent this block, no?

Reply:
No. `project_verison` may be passed explicitly. If not, then it's set from `config.version`. If it's still blank, *then* we're doing this block.

Reply:
I don't follow – that's exactly what would happen if you unindent?

Under what circumstances would the behavior differ if you unindented?

Reply:
The outer `if project_version is None` comes from the option passed to the `__main` click command.
The inner `if project_version is None` comes from `config.version`
We only want to try and get the version if both of these are null, *and* if there's a `config.package` actually set (otherwise we raise the new `UsageError`).

Reply:
The only difference between this code and the previous code is the new exception and condition. It's just moved to be in a more logical place in the command (i.e. before trying to load the template, etc).

Reply:
Maybe I was unclear (sorry on phone): I'm not asking you to remove any logic but to unindent the whole block starting with the second `if project_version is None:`.

AFAICT, this is the classic "try to fill a variable from different sources until it's not None" pattern.

Reply:
I think I was just being dense. I get you now 🤣